contrib/coredns/policy/policy_test.go

```diff
@@ -0,0 +1,105 @@
+package policy
+
+import (
+	"testing"
+	"errors"
+	"github.com/miekg/dns"
+	"net"
+
+)
+
+
+
+// This function tests that Validation of Middleware response is working properly
+// Inputs are t *testing.T
+// Output is the error update in t
+func TestValidateMiddlewareResponse(t *testing.T) {
+	var p PolicyMiddleware
+	var dnsResponseWriter	dns.ResponseWriter
+
+	dnsMsg := new(dns.Msg)
+	//
+	// Case 1
+	//
+	var response1 Response
+
+	// Case 1a - Err is set, return should have err and RcodeServerFailure
+	err := errors.New("Invalid error")
+	status := dns.RcodeServerFailure
+	returnedStatus, returnedError := ValidateMiddlewareResponse(&p, dnsResponseWriter, dnsMsg, &response1, status, err)
```

Comment:
should still go in that file, we will just be testing the ServeDNS method of the policy middleware